### ISSUE TITLE
BaseReferance -> straxen.get_resources incompatible with tar.gz format (tensorflow models for posrec)

### ISSUE THREAD

**Issue by LuisSanchez25** [XENONnT/xedocs#17]
https://github.com/XENONnT/xedocs/blob/2693e075681263164cf9804bafe84e04009ff7ac/xedocs/schemas/corrections/base_references.py#L82

Minimal working example:
```
xedocs.PosRecModel(kind = 'mlp', time = (start_time, end_time), 
               version = 'v!!', value = 'xnt_mlp_wfsim_20201214.tar.gz', fmt='abs_path').save()

```
Raises error:
'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte.

This issue is due to the get_resources function in straxen as it does not have a way to handle tar files. We could either patch get resources or modify the schema for posrec.

**Comment by jmosbacher** [comment]
@LuisSanchez25 this has nothing to with `get_resource` not being compatible with tar.gz
There are multiple issues here:
1) you are trying to use `fmt=abs_path` but that argument is not a valid fmt argument to `get_resource`, its a valid argument to the `resource://` protocol which results in bypassing the `get_resource` method and fetching the file directly using the downloader. 
2) you are setting `fmt` on the document but its actually not defined as a field in the schema, its defined as a class attribute so passing it to the constructor is not doing anything and the fmt remains its default value of json.

We can change fmt to be a field in the schema so that it can be set on a per-document basis. this way each document in the DB can have a different fmt but im not sure we want to allow this. It can make problems if you have the fmt changing over time .

So in summary the schema actually behaved as expected. The accepted format of this schema is json and you tried to insert a tar.gz so it raised an error.
